### ISSUE TITLE
Riot: clarify the 'deactivate account' UX

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-meta#164]

**Comment by ara4n** [comment]
I don't think we even have a deactivate button on iOS or Android today, which obviously should be added as part of this.

This has two parts:
 * [ ] Improve wording explaining what deactivation means
 * [ ] Add a checkbox to request a GDPR erasure at the point of deactivation, which should then hit whatever the API synapse provides for GDPR erasure will be.

**Comment by manuroe** [comment]
The current UI/UX for account deactivation from the settings page on riot-web is:

<img width="613" alt="screen shot 2018-05-22 at 07 44 55" src="https://user-images.githubusercontent.com/8418515/40344271-56cb6588-5d94-11e8-9be6-f2e3a9a438ff.png">

**Comment by manuroe** [comment]
The CS API endpoint is https://matrix.org/docs/spec/client_server/r0.3.0.html#post-matrix-client-r0-account-deactivate.

**Comment by lukebarnard1** [comment]
Most up-to-date wording spec is https://docs.google.com/document/d/1n0YmfdT8y28cVyWkRzeradSFKZrX8lYADhFBUbGU_LQ/edit#

**Comment by manuroe** [comment]
@SBiOSoftWhare , @bmarty , here is how the erase option is implemented in matrix-js-sdk: https://github.com/matrix-org/matrix-js-sdk/pull/649/files#diff-7ecbd1742c9fa46478efe534d035eb28R341
